tools/cli/air-trace/README.md

```diff
@@ -0,0 +1,50 @@
+# The `air-trace` utility
+
+The main propose of this utility is to run AquaVM with data and analyze its performance.  It has two subcommands: `run` and `stats`.
+
+## `air-trace run`
+
+Executes an AIR script with data in WASM AquaVM.  It has two modes of parameter input: plain and anomaly.
+
+### Common parameters
+All common parameters are optional.  Their position is before the mode selector (`--plain` or `--anomaly`).
+
++ `--call_results PATH` parameter allows you to provide call results for current execution.
++ `--current-peer-id STR` by default is "some_id".
++ `--max-heap-size N` defines maximum heap size for WASM runtime.
++ `--runtime PATH` option defines the AquaVM WASM binary to be executed.  By default, it is "target/wasm32-wasi/release/air_interpreter_server.wasm", but you can define a global value with the `AIR_WASM_RUNTIME_PATH` environment variable.
++ with the `--json` option, tracing info is output in machine-readable JSON format.  The output can be later processed with `air-trace stats` subcommand.
++ `--tracing-params` defines tracing crate logging levels.  By default, it is equal to `info` and does trace the most high-level AquaVM constructions (data parsing, AIR script parsing, execution, result construction).  With `debug` level it traces some individual commands, and with `trace` level it traces event more fine grained functionality.
+
+The important option is `--native`.  It runs the AquaVM as the native code that can be profiled with any native profiler.  As input data deserialization and serialization time can be comparable to particle execution time, and short execution times provide less reliable results, one can use `--repeat N` option to repeat particle execution several times.  Execution result is not printed in this case, so you may run `--repeat 1` to suppress it.
+
+Run `air-trace run --help` to see all common parameters.
+
+### Plain mode
+In the `--plain` mode, the parameters like AIR script path, data path, previous data path and other particle fields can be provided in separate arguments, of which only `--data` is the required one (and AIR script is read from stdin by default).
+
+Run `air-trace run --plain --help` to see all plain mode options.
+
+### Anomaly mode
+In the anomaly mode, the only argument is a path to self-contained anomaly data file obtained from Anomaly Particle Detection System.
+
+Run `air-trace run --anomaly --help` to see all anomaly mode options.
+
+## `air-trace stats`
+
+This subcommand allows to process JSON trace collected with `air-trace run --json`.  It has two primary options:
+
++ `--pretty` outputs JSON trace in human readable format.
++ `--stats` outputs execution summary.
+
+By default, both options are effective.
```

Comment:
If you provide options explicitly, only provided options are effective.  So, just run `air-trace stats --stats` to see summary.

BTW, it might be renamed to `--summary`.

Reply:
`--summary` is better. 
> If you provide options explicitly, only provided options are effective. 
I don't think it is good design. I think better have options for disabling default behavior. Like, `--pretty` +  `--unformatted/--no-pretty`. Right now effect of `--stats/summary` option is "disable pretty-printing", which has nothing in common with the name.

Reply:
They are orthogonal, but default includes both of them just for convenience. Is there any other useful behavior for `stats` without arguments?

Reply:
Then I think it must be three options for all variants. Because I understand `--pretty` as "how to print data" and "--summary" as "what data to print". In that case, they must not influence each other. But if "--pretty" is to just pretty-print json, then I don't understand why to run air-trace for it. You can use "jq --pretty" with the same effect.

Reply:
It does not output human-readable JSON, it outputs human-readable logs (not in JSON form).

There is no point of introducing third "cake and tea" option, as cake and tea are orthogonal, they are just different sections of the report produced.  Do you want first section, second section or both?